src/Admin/DownloadPaths/class-dlm-downloads-path.php

```diff
@@ -0,0 +1,559 @@
+	 * @since 5.0.0
+	 */
+	public static function get_instance() {
+		if ( ! isset( self::$instance ) && ! ( self::$instance instanceof DLM_Downloads_Path ) ) {
+			self::$instance = new DLM_Downloads_Path();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Set required admin hooks for the Download Monitor settings.
+	 *
+	 * @since 5.0.0
+	 */
+	private function set_admin_hooks() {
+		if ( ! is_admin() ) {
+			return;
+		}
+		// Add Templates tab in the Download Monitor's settings page.
+		add_filter( 'dlm_settings', array( $this, 'status_tab' ), 15, 1 );
+		// Show the approved downloads path tab content.
+		add_action( 'dlm_tab_section_content_download_path', array( $this, 'paths_content' ) );
+		add_action( 'admin_init', array( $this, 'register_setting' ) );
+		add_filter( 'pre_update_option', array( $this, 'update_action' ), 10, 3 );
+		add_action( 'admin_init', array( $this, 'actions_handler' ) );
+		add_action( 'admin_init', array( $this, 'bulk_actions_handler' ) );
+		// Hide the save button in the Approved Download Paths tab.
+		add_filter( 'dlm_show_save_settings_button', array( $this, 'hide_save_button' ), 15, 3 );
+	}
+
+	/**
+	 * Set required admin hooks for the Download Monitor settings.
+	 *
+	 * @since 5.0.0
+	 */
+	private function set_frontend_hooks() {
+		if ( is_admin() ) {
+			return;
+		}
+		// We need to set the setting for the frontend as well, as we need the default return value.
+		add_action( 'init', array( $this, 'register_setting' ) );
+	}
+
+	/**
+	 * Register settings for advanced download path.
+	 *
+	 * @since 5.0.0
+	 */
+	public function register_setting() {
+		// Register the setting for multisite.
+		if ( is_multisite() ) {
+			$multi_args = array(
+				'type'    => 'array',
+				'default' => array(
+					'dlm_turn_off_file_browser' => '0',
+				),
+			);
+			register_setting( 'dlm_advanced_download_path', 'dlm_network_settings', $multi_args );
+
+			if ( ! empty( $_GET['id'] ) && ! empty( $_GET['page'] ) && 'download-monitor-paths' === $_GET['page'] ) {
+				$site_id = absint( $_GET['id'] );
+				// Create the uploads path for the site.
+				$uploads = WP_CONTENT_DIR . '/uploads/sites/' . $site_id;
+
+				$default = array(
+					array(
+						'id'       => 2,
+						'path_val' => trailingslashit( $uploads ),
+						'enabled'  => true,
+					),
+				);
+				$args    = array(
+					'type'    => 'array',
+					'default' => $default,
+				);
+
+				register_setting( 'dlm_advanced_download_path', 'dlm_allowed_paths', $args );
+			} else {
+				$site_id = get_current_blog_id();
+				// Create the uploads path for the site.
+				$uploads = WP_CONTENT_DIR . '/uploads/sites/' . $site_id;
+
+				$default = array(
+					array(
+						'id'       => 2,
+						'path_val' => trailingslashit( $uploads ),
+						'enabled'  => true,
+					),
+				);
+				$args    = array(
+					'type'    => 'array',
+					'default' => $default,
+				);
+				register_setting( 'dlm_advanced_download_path', 'dlm_allowed_paths', $args );
+			}
```

Comment:
mult cod repetat si labartat, e f greu de citit - poate fi simplificat
```suggestion
			$site_id = get_current_blog_id();
			$uploads = WP_CONTENT_DIR . '/uploads/sites/' . $site_id;
			if ( ! empty( $_GET['id'] ) && ! empty( $_GET['page'] ) && 'download-monitor-paths' === $_GET['page'] ) {
				$site_id = absint( $_GET['id'] );
				// Create the uploads path for the site.
				$uploads = WP_CONTENT_DIR . '/uploads/sites/' . $site_id;
			} 
			$default = array(
				array(
					'id'       => 2,
					'path_val' => trailingslashit( $uploads ),
					'enabled'  => true,
				),
			);
			$args    = array(
				'type'    => 'array',
				'default' => $default,
			);

			register_setting( 'dlm_advanced_download_path', 'dlm_allowed_paths', $args );
```